java/jdbc/SpringBootSample/src/main/java/com/oracle/springapp/dao/impl/AllTablesDAOImpl.java

```diff
@@ -0,0 +1,43 @@
+package com.oracle.springapp.dao.impl;
+
+import java.util.List;
+
+import javax.annotation.PostConstruct;
+import javax.sql.DataSource;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.jdbc.core.support.JdbcDaoSupport;
+import org.springframework.stereotype.Repository;
+
+import com.oracle.springapp.dao.AllTablesDAO;
+import com.oracle.springapp.model.AllTables;
+/**
+ * Simple Java class which uses Spring's JdbcDaoSupport class to implement
+ * business logic.
+ *
+ */
+@Repository
+public class AllTablesDAOImpl extends JdbcDaoSupport implements AllTablesDAO {
+	@Autowired
+	private DataSource dataSource;
+
+	@PostConstruct
+	public void initialize() {
+		setDataSource(dataSource);
+		System.out.println("Datasource used: " + dataSource);
+	}
+
+	@Override
+	public List<AllTables> getTableNames() {
+		final String sql = "SELECT owner, table_name, status, num_rows FROM all_tables where rownum < 20";
```

Comment:
Please use a PreparedStatement with a bind variable; this is the best practice we have been preaching.